### ISSUE TITLE
Duplicate JSESSIONID sent when invalidating new session

### ISSUE THREAD

**Issue by stuban** [eclipse/jetty.project#3421]
When invalidating a (new) session and immediately creating a new when jetty 9.4.x sends the JSESSION cookie twice which confuses some browsers.
This seems to be a regression as it did behave differently in 9.2.x
Example code:
```java
@WebServlet("/sessionTest")
public class SessionServlet extends HttpServlet {

    @Override
    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
        resp.setContentType("text/html;charset=UTF-8");
        PrintWriter out = resp.getWriter(); 
        out.println("<html lang=\"en\">" + "<head><title>Servlet Hello</title></head>");
        out.println("<body><h2>Hello World!</h2>");
        out.println("<p>Current sessionID:" + (req.getSession(false) == null ? "empty" : req.getSession(false).getId())+"</p>");
        out.println("<p><form method=\"post\"><input type=\"submit\" value=\"New Session please\">");
        out.println("</form>");
        out.println("</body></html>");
        
    }
    
    @Override
    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
        req.getSession().invalidate();
        req.getSession().setAttribute("testKey", "testValue");
        resp.setContentType("text/html;charset=UTF-8");
        PrintWriter out = resp.getWriter();
        out.println("<html lang=\"en\">" + "<head><title>Servlet Hello</title></head>");
        out.println("<body><h2>Tried to create a new session</h2>");
        out.println("<p>Current sessionID:" + (req.getSession(false) == null ? "empty" : req.getSession(false).getId())+"</p>");
        out.println("<p><form method=\"post\"><input type=\"submit\" value=\"New Session please\">");
        out.println("</form>");
        out.println("</body></html>");
     
    }
}
```

Output from 9.2.26:
```
POST /test/sessionTest HTTP/1.1
Host: localhost:8080
User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:65.0) Gecko/20100101 Firefox/65.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
Accept-Language: en,en-US;q=0.95,de;q=0.89,fr-FR;q=0.84,nl;q=0.79,de-DE;q=0.74,en-GB;q=0.68,de-US;q=0.63,en-EN;q=0.58,fr;q=0.53,nl-NL;q=0.47,de-formal;q=0.42,en-CA;q=0.37,en-AU;q=0.32,es-ES;q=0.26,es;q=0.21,en-DE;q=0.16,eng-US;q=0.11,eng;q=0.05
Accept-Encoding: gzip, deflate
Referer: http://localhost:8080/test/sessionTest
Content-Type: application/x-www-form-urlencoded
Content-Length: 0
DNT: 1
Connection: keep-alive
Upgrade-Insecure-Requests: 1

HTTP/1.1 200 OK
Date: Mon, 04 Mar 2019 09:46:26 GMT
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Set-Cookie: JSESSIONID=1fo6m5u049z11sb9uxibgs2ja;Path=/test
Content-Type: text/html;charset=UTF-8
Content-Length: 249
Server: Jetty(9.2.26.v20180806)

<html lang="en"><head><title>Servlet Hello</title></head>
<body><h2>Tried to create a new session</h2>
<p>Current sessionID:1fo6m5u049z11sb9uxibgs2ja</p>
<p><form method="post"><input type="submit" value="New Session please">
</form>
</body></html>
```

Output from 9.4.15:
```
POST /test/sessionTest HTTP/1.1
Host: localhost:8080
User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:65.0) Gecko/20100101 Firefox/65.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
Accept-Language: en,en-US;q=0.95,de;q=0.89,fr-FR;q=0.84,nl;q=0.79,de-DE;q=0.74,en-GB;q=0.68,de-US;q=0.63,en-EN;q=0.58,fr;q=0.53,nl-NL;q=0.47,de-formal;q=0.42,en-CA;q=0.37,en-AU;q=0.32,es-ES;q=0.26,es;q=0.21,en-DE;q=0.16,eng-US;q=0.11,eng;q=0.05
Accept-Encoding: gzip, deflate
Referer: http://localhost:8080/test/sessionTest
Content-Type: application/x-www-form-urlencoded
Content-Length: 0
DNT: 1
Connection: keep-alive
Upgrade-Insecure-Requests: 1

HTTP/1.1 200 OK
Date: Mon, 04 Mar 2019 09:48:22 GMT
Set-Cookie: JSESSIONID=node094iqyxoe2z8h7ojaybhzk30g0.node0;Path=/test
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Set-Cookie: JSESSIONID=node0tyopewww0zyps62cnw8h33ti1.node0;Path=/test
Content-Type: text/html;charset=utf-8
Content-Length: 254
Server: Jetty(9.4.15.v20190215)

<html lang="en"><head><title>Servlet Hello</title></head>
<body><h2>Tried to create a new session</h2>
<p>Current sessionID:node0tyopewww0zyps62cnw8h33ti1</p>
<p><form method="post"><input type="submit" value="New Session please">
</form>
</body></html>
```

This only happens when there has been no session before so the line req.getSession().invalidate(); creates and invalidates a session in one go.

**Comment by janbartel** [comment]
@stuban  you're absolutely right. Looks like the cookie handling code got changed and no longer replaces a cookie of the same name.  Fix underway.

**Comment by janbartel** [comment]
Fix applied to jetty-9.4.x for jetty-9.4.16.
